src/main/java/uk/gov/justice/hmpps/prison/api/resource/KeyWorkerResource.java

```diff
@@ -39,74 +40,52 @@ public KeyWorkerResource(final KeyWorkerAllocationService keyWorkerService) {
+        @ApiResponse(responseCode = "404", description = "Requested resource not found.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
+        @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
     @Operation(summary = "Key workers available for allocation at specified agency.", description = "Key workers available for allocation at specified agency.")
     @GetMapping("/{agencyId}/available")
+    @VerifyAgencyAccess(overrideRoles = {"KEY_WORKER"})
     @SlowReportQuery
     public List<Keyworker> getAvailableKeyworkers(@PathVariable("agencyId") @Parameter(description = "The agency (prison) identifier.", required = true) final String agencyId) {
         return keyWorkerService.getAvailableKeyworkers(agencyId);
     }
 
     @ApiResponses({
-            @ApiResponse(responseCode = "200", description = "OK"),
-            @ApiResponse(responseCode = "400", description = "Invalid request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
-            @ApiResponse(responseCode = "404", description = "Requested resource not found.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
-            @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
-    @Operation(summary = "Specified key worker's currently assigned offenders.", description = "Specified key worker's currently assigned offenders.")
-    @GetMapping("/{staffId}/agency/{agencyId}/offenders")
-    public List<KeyWorkerAllocationDetail> getAllocationsForKeyworker(@PathVariable("staffId") @Parameter(description = "The key worker staff id", required = true) final Long staffId, @PathVariable("agencyId") @Parameter(description = "The agency (prison) identifier.", required = true) final String agencyId) {
-        return keyWorkerService.getAllocationDetailsForKeyworkers(Collections.singletonList(staffId), agencyId);
-    }
-
-    @ApiResponses({
-            @ApiResponse(responseCode = "200", description = "The allocations list is returned.")})
+        @ApiResponse(responseCode = "200", description = "The allocations list is returned.")})
     @Operation(summary = "Retrieves Specified key worker's currently assigned offenders - POST version to allow larger staff lists.", description = "Retrieves Specified key worker's currently assigned offenders - POST version to allow larger staff lists.")
     @PostMapping("/{agencyId}/current-allocations")
+    // @PreAuthorize("hasRole('KEY_WORKER')")
+    @SlowReportQuery
```

Comment:
Client (keyworker_api) is currently calling these endpoint with a user/auth token. Leave the @ PreAuthorize until client has switched to a system token